inst/tests/tests.Rraw

```diff
@@ -3221,6 +3221,10 @@ test(1034, as.data.table(x<-as.character(sample(letters, 5))), data.table(V1=x))
   test(1035.12, attr(melt(DT, id.vars=1:2)$x, "foo"), "bla1")
   test(1035.13, attr(melt(DT, id.vars=1:2)$y, "bar"), 1:4)
 
+  # issue #6867 - id.vars=patterns().
+  DT=data.table(x_long=0, x_short=0, z=0, y1=1, y2=2)
+  test(1035.131, melt(DT, measure.vars=patterns("y"), id.vars=patterns("x")), data.table(x_long=0, x_short=0, variable=factor(c("y1","y2")), value=c(1,2)))
```

Comment:
should this get more tests, e.g. error cases, cases where `id.vars`+`measure.vars` interact, or do you think that's well-covered enough by existing tests already?

(especially if using a helper, we are more confident tests applying to edge cases of `measure.vars` apply as well to `id.vars`).